### PR TITLE
[CIAS30-3685] Make the both ends of data range in pie charts inclusive

### PR DESCRIPTION
**Pull request by OskarDybowskiHTD**
## Related tasks
- [CIAS-3685](https://htdevelopers.atlassian.net/browse/CIAS30-3685)

## What's new?
- Chart statistics will be now generated assuming the date_range is from the very beginning of the first day to the very end of the last day

### CODE REVIEW

File: app/services/v1/chart_statistics/create.rb [hunk 1]
```diff
@@ -71,8 +71,8 @@ def zero_division_error?
   end
 
   def inside_date_range?
-    return false if chart.date_range_start.present? && chart.date_range_start > user_session.finished_at
-    return false if chart.date_range_end.present? && chart.date_range_end < user_session.finished_at
+    return false if chart.date_range_start.present? && chart.date_range_start.beginning_of_day > user_session.finished_at
```
Comment (WiktoriaRozanska) [inline app/services/v1/chart_statistics/create.rb:74]: why not `>=` ?

File: app/services/v1/chart_statistics/create.rb [hunk 2]
```diff
@@ -72,7 +72,7 @@ def zero_division_error?
 
   def inside_date_range?
     return false if chart.date_range_start.present? && chart.date_range_start > user_session.finished_at
-    return false if chart.date_range_end.present? && chart.date_range_end < user_session.finished_at
+    return false if chart.date_range_end.present? && chart.date_range_end + 1.day <= user_session.finished_at
```
Comment (OskarDybowskiHTD) [inline app/services/v1/chart_statistics/create.rb:75]: FE sends the *beginning* of the last day in the date range, so we need to offset it by 1

Comment (WiktoriaRozanska) [reply app/services/v1/chart_statistics/create.rb:75]: so why not chart.date_range_end.end_of_day?

Comment (WiktoriaRozanska) [reply app/services/v1/chart_statistics/create.rb:75]: I think it is more intuitive

Comment (OskarDybowskiHTD) [reply app/services/v1/chart_statistics/create.rb:75]: The timestamp FE sends doesn't include the timezone, and I have agreed with @bartosz-kepka-htd that's going to be the best solution

**Review by WiktoriaRozanska — APPROVED** [review]

**Pull request merged**